src/client.rs

```diff
@@ -244,7 +232,7 @@ impl Spotify {
     /// - market - an ISO 3166-1 alpha-2 country code.
     pub async fn tracks(
         &self,
-        track_ids: Vec<&str>,
+        track_ids: impl IntoIterator<Item = &str>,
```

Comment:
One thing I've realized is the inconsistency with `impl IntoIterator<Item = &str>`. The changelog indicates that this is the new function signature for iterable parameters, but sometimes you use `impl IntoIterator<Item = String>` or `impl IntoIterator<Item = &String>`. I would go with the `&str` solution to avoid unnecessary allocations, and only use that type in this module.

Reply:
> One thing I've realized is the inconsistency with impl IntoIterator<Item = &str>. The changelog indicates that this is the new function signature for iterable parameters

Yes, I have noticed the `inconsistency` with `impl IntoIterator<Item = &str>`, but it's caused by the original design, not the introduction of `impl IntoIterator<Item = &str>`. 

> but sometimes you use impl IntoIterator<Item = String> or impl IntoIterator<Item = &String>

Because some functions use `&[String]` and some functions use `Vec<&str>`, If I change them to `impl IntoIterator<Item = &String>`, it will break up the backward compatibility.

> I would go with the &str solution to avoid unnecessary allocations, and only use that type in this module.

I have thinked about this solution, it's fine to break up backward compatibility at this release.

Reply:
I would say it's OK to break backward compatibility with this. We're going to change lots of other things anyway.